File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/kafkakeys/AggregateKey.java
package com.kafkaquest.kq.common.util.models.kafkakeys;

import com.kafkaquest.kq.common.util.models.events.enums.EventType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AggregateKey {

    private EventType eventType;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Keys/AggregateKey.cs
namespace KQ_Projection.Common.Models.Keys;

using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

public class AggregateKey
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventType eventType { get; set; }
}
